doc/development_process/api_lifecycle.rst

```diff
@@ -77,16 +77,70 @@ An API can be declared ``stable`` after fulfilling the following requirements:
 - Stable APIs can get backward compatible updates, bug fixes and security fixes
   at any time.
 
+Introducing incompatible changes
+================================
+
+A stable API, as described above strives to remain backwards-compatible through
+its life-cycle. There are however cases where fulfilling this objective prevents
+technical progress or is simply unfeasible without unreasonable burden on the
+maintenance of the API and its implementation(s).
+
+In order to restrict and control the introduction of a change that breaks the
+promise of backwards compatibility the following steps must be followed whenever
+such a change is considered necessary in order to accept it in the project:
+
+#. An RFC issue must be opened on GitHub with the following content:
+
+   .. code-block:: none
+
+      Title:     RFC: API Change: <subsystem>
+      Contents:  - Background information
+                 - Details on the proposed change
+                   - Function call changes
+                   - Device Tree changes (source and bindings)
+                   - Kconfig option changes
+                 - Impact to users of the API, including the steps required
+                   to adapt out-of-tree users of the API to the change
+
+#. A Pull Request must be opened on GitHub that includes the following:
+
+   - A title that matches the RFC issue
+   - The actual changes to the API:
+     - Changes to the API header file
+     - Changes to the API implementation(s)
+     - Changes to the relevant API documentation
+     - Changes to Device Tree source and bindings
+   - The changes required to adapt all in-tree users of the API to the change
+   - An entry in the "API Changes" section of the release notes for the next
+     upcoming release
+   - The labels ``API`` and ``API change``
+
+#. An email must be sent to the ``tsc``, ``users`` and ``devel`` mailing lists
+   with a subject identical to the RFC issue title and that links to both the
+   RFC issue and the Pull Request
+
+Once the steps above have been completed, the outcome of the proposal will
+depend on the approval of the actual Pull Request by the maintainer of the
+corresponding subystem. As with any other Pull Request, the author can request
+for it to be discussed and ultimately even voted on in any of the regularly
```

Comment:
add a link:
```
in any of the `regularly scheduled meetings
<https://wiki.zephyrproject.org/Zephyr-Committee-and-Working-Group-Meetings>`_.
```

Reply:
Can we just have people send the email to the devel@ please? This is not an announcement, this just someone opening an issue.